src/data_info/services/rename.clj

```diff
@@ -19,6 +21,42 @@
   [source-path dest-path]
   (ft/path-join dest-path (ft/basename source-path)))
 
+(defn validate-unlocked
+  [paths]
+  ;; We need to:
+  ;; Fetch all async tasks that could cause it to be locked (data-move, data-rename, data-delete, data-delete-trash, data-restore)
+  ;; statuses registered, started, running, detected-stalled (so, not completed/failed mostly)
+  ;; or we can just only include non-completed ones, i.e. EndDateSince sometime in the future + include null end
+  ;; Pull out their paths.
+  ;; data-move: sources, destination
+  ;; data-rename: source, destination
+  ;; data-delete: paths, trash-paths (not that the latter is real likely)
+  ;; data-delete-trash: trash-paths
+  ;; data-restore: paths, restoration-paths
+  ;; Then we need to check all of `paths` to see if any of them matches any of the paths we fetched. These should either match exactly, or match something up through a slash in the path passed to us (i.e., a folder -- we don't want to lock /x/y/zarbee if someone is moving /x/y/za to /x/y/zb or whatever)
+  (let [far-future "9999-12-31T23:59:59Z"
```

Comment:
Glad I won't be around for the inevitable Y10K apocalypse. 😁

Reply:
I figure, if anyone is still running this code unmodified at that point, they have bigger problems :laughing: